pact/Pact/Core/Errors.hs

```diff
@@ -1061,6 +1095,586 @@ deriveConstrInfo ''EvalError
+  EvalError txt ->
+    thsep ["Program encountered an unhandled raised error:", txt]
+  YieldOutsideDefPact ->
+    "Try to yield a value outside a running DefPact execution"
+  NoActiveDefPactExec ->
+    "No active DefPact execution in the environment"
+  NoYieldInDefPactStep (DefPactStep step _ (DefPactId dpid) _) ->
+    thsep ["No yield in DefPactStep.", "Step:", tInt step, "DefPactId:", dpid]
+  InvalidDefPactStepSupplied (DefPactStep step _ _ _) stepCount ->
+    thsep
+    [ "DefPactStep does not match DefPact properties:"
+    , "requested:", tInt step
+    , "step count:", tInt stepCount]
+  DefPactIdMismatch (DefPactId reqId) (DefPactId envId) ->
+    thsep
+    [ "Requested DefPactId:", reqId
+    , "does not match context DefPactId:", envId
+    ]
+  CCDefPactContinuationError pactStep _ccExec _dbExec ->
+    thsep
+    [ "Crosschain DefPact continuation error:"
+    , "DefPactId:", (tDpId (_psDefPactId pactStep))
+    ]
+  NestedDefPactParentRollbackMismatch pid rollback parentRollback ->
+    thsep
+    [ "Nested DefPact execution failed, parameter missmatch:"
+    , "DefPactId:", tDpId pid
+    , "Rollback:", tBool rollback
+    , "Parent rollback:", tBool parentRollback
+    ]
+  NestedDefPactParentStepCountMismatch pid stepCount parentStepCount ->
+    thsep
+    [ "Nested DefPact execution failed, parameter missmatch:"
+    , "PactId:", tDpId pid
+    , "step count:", T.pack (show stepCount)
+    , "Parent step count:", T.pack (show parentStepCount)
+    ]
+  NoPreviousDefPactExecutionFound ps ->
+    thsep ["No previous DefPact execution found for DefPactId: ", tDpId (_psDefPactId ps)]
+  DefPactAlreadyCompleted ps -> thsep
+    [ "Requested DefPact already completed:", "DefPactId:", tDpId (_psDefPactId ps)]
+  NestedDefPactNeverStarted ps -> thsep
+    ["Requested nested DefPact never started:", "DefPactId:", tDpId(_psDefPactId ps)]
+  NestedDefPactDoubleExecution ps -> thsep
+    ["Requested nested DefPact double execution:", "DefPactId:" , tDpId (_psDefPactId ps)]
+  MultipleOrNestedDefPactExecFound pe -> thsep
+    ["DefPact execution context already in the environment: DefPactId:", tDpId (_peDefPactId pe)]
+  DefPactStepHasNoRollback ps -> thsep
+    ["Step has no rollback:", "DefPactId:", tDpId (_psDefPactId ps)]
+  DefPactStepNotInEnvironment -> "No DefPactStep in the environment"
+  NoDefPactIdAndExecEnvSupplied -> "No DefPactId or execution environment supplied"
+  DefPactRollbackMismatch ps pe -> thsep
+    [ "Rollback missmatch in DefPactStep and DefPact execution environment:"
+    , "DefPactId:", tDpId (_psDefPactId ps)
+    , "step rollback:", tBool (_psRollback ps)
+    , "DefPactExec rollback:", tBool (_peStepHasRollback pe)
+    ]
+  DefPactStepMismatch ps pe -> thsep
+    [ "Step missmatch in DefPactStep and DefPact execution environment:"
+    , "DefPactId:", tDpId (_psDefPactId ps)
+    , "step:", tInt (_psStep ps)
+    , "DefPactExec step:", tInt (_peStep pe + 1)
+    ]
```

Comment:
Maybe this is just because I don't know much about nested defpacts, but I'm not sure about all of these, except "already completed", "never started", and "double execution" (which maybe should be "already executed")